src/core/stdc/errno.d

```diff
@@ -75,14 +75,6 @@ else version (Darwin)
         alias errno = __error;
     }
 }
-else version (OSX)
-{
-    extern (C)
-    {
-        ref int __error();
-        alias errno = __error;
-    }
-}
 else
 {
```

Comment:
One thing I noticed is if I tried to go back to this old default block by making sure none of the above versions hit, I got an error in [`core.memory` here about "errno is not an lvalue"](https://github.com/dlang/druntime/blob/master/src/core/memory.d#L908), so it appears #1917 can't be used with this default block anymore.